src/expr.cpp

```diff
@@ -2647,6 +2647,43 @@ void BinaryExpr::Print() const {
     pos.Print();
 }
 
+llvm::Constant *BinaryExpr::GetConstant(const Type *type) const {
+
+    // Are we doing something like (basePtr + offset)[...] = ... for a Global
+    // Variable
+    if (!GetLValueType())
+        return NULL;
+
+    if (!((op == BinaryExpr::Op::Add) || (op == BinaryExpr::Op::Sub)))
+        return NULL;
+    else if (op == BinaryExpr::Op::Sub) {
+        if (const PointerType *pt1 = CastType<PointerType>(arg1->GetType())) {
```

Comment:
Why defining and assigning to pt1 here? It's confusing and makes the intentions not clear.
Also adding comment why this case is rejects will not harm.

Reply:
Done